File: backend/app/schemas/ticket.py
from pydantic import BaseModel


class TicketRequest(BaseModel):
    ticket: str
    reply_language: str


class TicketAnalysis(BaseModel):
    category: str
    priority: str
    priority_reason: str
    confidence_score: int
    confidence_reason: str
    problem_summary: str
    likely_causes: list[str]
    troubleshooting_steps: list[str]
    powershell_commands: list[str]
    security_notes: list[str]
    information_needed: list[str]
    estimated_resolution_time: str
    user_reply: str

File: frontend/src/types/ticket.ts
export type ReplyLanguage = "German" | "English";

export type TicketAnalysis = {
  category: string;
  priority: string;
  priority_reason: string;
  confidence_score: number;
  confidence_reason: string;
  problem_summary: string;
  likely_causes: string[];
  troubleshooting_steps: string[];
  powershell_commands: string[];
  security_notes: string[];
  information_needed: string[];
  estimated_resolution_time: string;
  user_reply: string;
};